chronos/apps/http/httpserver.nim

```diff
@@ -246,9 +246,12 @@ proc dumbResponse*(): HttpResponseRef {.raises: [Defect].} =
   ## Create an empty response to return when request processor got no request.
   HttpResponseRef(state: HttpResponseState.Dumb, version: HttpVersion11)
 
-proc getId(transp: StreamTransport): string {.inline.} =
+proc getId(transp: StreamTransport): Result[string, string]  {.inline.} =
   ## Returns string unique transport's identifier as string.
-  $transp.remoteAddress() & "_" & $transp.localAddress()
+  try:
+    ok($transp.remoteAddress() & "_" & $transp.localAddress())
+  except TransportOsError as exc:
+    err($exc.msg)
```

Comment:
`EINVAL` for this functions could have different meaning, so i wish not to rename it to `AbortedError`. And there is no need to distinguish because if you unable to identify local or remote side of socket you definitely lost this socket... So doesn't mater at what condition you are you should close the socket. Also `TransportOsError` ins this case should be handled as it handled for `send/receive` (which also could raise `TransportOsError`). Its not a critical error.